docs/design/sql/20-match-jet-job-to-sql.md

```diff
@@ -0,0 +1,92 @@
+# Matching the SQL query to the Jet job
+
+|||
+|---|---|
+|Related Jira|[HZ-2859](https://hazelcast.atlassian.net/browse/HZ-2859)|
+|Document Status / Completeness|IN PROGRESS|
+|Requirement owner|Saumitra Sahi|
+|Developer(s)|Sasha Syrotenko, Ivan Yaschishin|
+|Quality Engineer|Isaac Sumner|
+|Technical Reviewers|Krzystof Jamroz|
+|Version|5.4|
+
+## Background
+In Hazelcast Platform 5.3, the sole SQL job metrics report can be accessed exclusively through
+the `getJobAndSqlSummaryList()` method within the Java client. This approach doesn't align with
+the idea of SQL serving as the primary Hazelcast interface. Consequently, there is currently
+no ability to monitor the status of a running interactive SQL query, because it's impossible
+to associate it automatically with the corresponding Jet job.
+
+To address this issue, we aim to enable the match of interactive SQL queries with their running Jet jobs,
+allowing to show live metrics for a query user just started.
+
+## Terminology
+**Query runtime** - query execution time; time between query submission and query completion.
+
+## Goals
+
+The goals of that initiative are:
+- open a possibility to capture live metrics by matching SQL query with running Jet job.
+- ability to cancel streaming query by cancelling the corresponding job.
+
+Non-goals are:
+- expose live metrics for non-Jet SQL query executions.
+- support`CREATE JOB` as part of this initiative. It is created with name, so it is already exposed.
+
+## Technical Design
+
+We can provide multiple approaches to solve given problem:
+
+### Reuse existing QueryId
+
+`QueryId` class appears to be a data structure for representing a query identifier. In fact, it is a 256-bit value,
```

Comment:
queryId is used only SELECT, how will that work for DML? Will you add DML to `QueryResultRegistry`? BTW, we might need to do that anyway in order to return correct number of rows affected by DML (MC wants to know that) instead of 0.

Reply:
see HZ-3585

Reply:
BTW, there are also in some dialects queries like `INSERT into <sth> select <sth> returning <ids>`. But if we want that, they could probably be treated like selects with side effects

Reply:
> BTW, we might need to do that anyway in order to return correct number of rows affected by DML (MC wants to know that) instead of 0.

That's actually a good point of introducing it, but for solving HZ-3585.